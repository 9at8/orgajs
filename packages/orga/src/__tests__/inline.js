const { parse } = require('../inline')

describe('Inline Parsing', () => {
  it('recon single emphasis', () => {

    const str = 'hello *world*, welcome to *org-mode*. Have a nice *day*.'
    expect(parse(str)).toMatchSnapshot()
  })

  it('recon mixed emphasis', () => {

    const str = `
[[https://github.com/xiaoxinghu/orgajs][Here's]] to the *crazy* ones, the /misfits/, the _rebels_, the ~troublemakers~, the round pegs in the +round+ square holes...
`
    expect(parse(str)).toMatchSnapshot()
  })

  it('recon emphasises at different locations', () => {
    expect(parse(`one *two* three`)).toMatchSnapshot()
    expect(parse(`*one* two three`)).toMatchSnapshot()
    expect(parse(`one two *three*`)).toMatchSnapshot()
  })

  it('recon link', () => {
    expect(parse(`hello [[./image/logo.png]]`)).toMatchSnapshot()
    expect(parse(`hello [[Internal Link][link]]`)).toMatchSnapshot()
    expect(parse(`hello [[../image/logo.png][logo]]`)).toMatchSnapshot()
  })

  it('recon footnote reference', () => {
    expect(parse(`hello[fn:1] world.`)).toMatchSnapshot()
  })

  it('recon emphasises with 2 chars', () => {
    expect(parse(`*12*`)).toMatchSnapshot()
    expect(parse(`*1*`)).toMatchSnapshot()
  })

  it('recon invalid inline markups', () => {
    [ ',', '\'', '"', ' ' ].forEach(c => {
      expect(parse(`*${c}word*`)).toMatchSnapshot() // begin
      expect(parse(`*word${c}*`)).toMatchSnapshot() // end
    })
  })
})
