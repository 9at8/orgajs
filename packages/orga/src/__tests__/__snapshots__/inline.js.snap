// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline Parsing can handle something more complicated 1`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "Special characters ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "~",
      },
    ],
    "type": "verbatim",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": " and ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "!",
      },
    ],
    "type": "verbatim",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": ". Also ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "~/.this/path",
      },
    ],
    "type": "verbatim",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": " and ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "that",
      },
    ],
    "type": "code",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": " thing.",
  },
]
`;

exports[`Inline Parsing recon emphasises at different locations 1`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "one ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "two",
      },
    ],
    "type": "bold",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": " three",
  },
]
`;

exports[`Inline Parsing recon emphasises at different locations 2`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "one",
      },
    ],
    "type": "bold",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": " two three",
  },
]
`;

exports[`Inline Parsing recon emphasises at different locations 3`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "one two ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "three",
      },
    ],
    "type": "bold",
  },
]
`;

exports[`Inline Parsing recon emphasises with 2 chars 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "12",
      },
    ],
    "type": "bold",
  },
]
`;

exports[`Inline Parsing recon emphasises with 2 chars 2`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "1",
      },
    ],
    "type": "bold",
  },
]
`;

exports[`Inline Parsing recon footnote reference 1`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "hello",
  },
  Object {
    "children": Array [],
    "label": "1",
    "type": "footnote.reference",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": " world.",
  },
]
`;

exports[`Inline Parsing recon invalid inline markups 1`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "*,word*",
  },
]
`;

exports[`Inline Parsing recon invalid inline markups 2`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "*word,*",
  },
]
`;

exports[`Inline Parsing recon invalid inline markups 3`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "*'word*",
  },
]
`;

exports[`Inline Parsing recon invalid inline markups 4`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "*word'*",
  },
]
`;

exports[`Inline Parsing recon invalid inline markups 5`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "*\\"word*",
  },
]
`;

exports[`Inline Parsing recon invalid inline markups 6`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "*word\\"*",
  },
]
`;

exports[`Inline Parsing recon invalid inline markups 7`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "* word*",
  },
]
`;

exports[`Inline Parsing recon invalid inline markups 8`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "*word *",
  },
]
`;

exports[`Inline Parsing recon link 1`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "hello ",
  },
  Object {
    "children": Array [],
    "desc": undefined,
    "type": "link",
    "uri": Object {
      "location": "./image/logo.png",
      "protocol": "file",
      "raw": "./image/logo.png",
    },
  },
]
`;

exports[`Inline Parsing recon link 2`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "hello ",
  },
  Object {
    "children": Array [],
    "desc": "link",
    "type": "link",
    "uri": Object {
      "location": "Internal Link",
      "protocol": "internal",
      "raw": "Internal Link",
    },
  },
]
`;

exports[`Inline Parsing recon link 3`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "hello ",
  },
  Object {
    "children": Array [],
    "desc": "logo",
    "type": "link",
    "uri": Object {
      "location": "../image/logo.png",
      "protocol": "file",
      "raw": "../image/logo.png",
    },
  },
]
`;

exports[`Inline Parsing recon mixed emphasis 1`] = `
Array [
  Object {
    "children": Array [],
    "desc": "Here's",
    "type": "link",
    "uri": Object {
      "location": "//github.com/xiaoxinghu/orgajs",
      "protocol": "https",
      "raw": "https://github.com/xiaoxinghu/orgajs",
    },
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": " to the ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "crazy",
      },
    ],
    "type": "bold",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": " ones, the ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "misfits",
      },
    ],
    "type": "italic",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": ", the ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "rebels",
      },
    ],
    "type": "underline",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": ", the ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "troublemakers",
      },
    ],
    "type": "code",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": ", the round pegs in the ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "round",
      },
    ],
    "type": "strikeThrough",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": " square holes...",
  },
]
`;

exports[`Inline Parsing recon single emphasis 1`] = `
Array [
  Object {
    "children": Array [],
    "type": "text",
    "value": "hello ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "world",
      },
    ],
    "type": "bold",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": ", welcome to ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "org-mode",
      },
    ],
    "type": "bold",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": ". Have a nice ",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "parent": [Circular],
        "type": "text",
        "value": "day",
      },
    ],
    "type": "bold",
  },
  Object {
    "children": Array [],
    "type": "text",
    "value": ".",
  },
]
`;
